### PR TITLE
Running LoRaWAN Network Server decoupled from IoT Edge

### PR DESCRIPTION
**Pull request by danigian**
# PR for issue #1553 

## What is being addressed

This PR is bringing into dev branch all the work related to [Decouple LNS from Edge milestone](https://github.com/Azure/iotedge-lorawan-starterkit/milestone/26)

## Validation:
- E2E Run with temporary changes to only target cloud deployment [here](https://github.com/Azure/iotedge-lorawan-starterkit/runs/7153938469) (the only failing test Test_OTAA_Confirmed_And_Unconfirmed_Message_With_Custom_RX1_DR_Offset_MultiGw is to be fixed by #1770)

### CODE REVIEW

File: LoRaEngine/LoraKeysManagerFacade/EdgeDeviceGetter.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        public EdgeDeviceGetter(IDeviceRegistryManager registryManager,
+                                ILoRaDeviceCacheStore cacheStore,
+                                ILogger<EdgeDeviceGetter> logger)
+        {
+            this.registryManager = registryManager;
+            this.cacheStore = cacheStore;
+            this.logger = logger;
+        }
+
+#pragma warning disable IDE0060 // Remove unused parameter. Kept here for future improvements of RegistryManager
```
Comment (Mandur) [inline LoRaEngine/LoraKeysManagerFacade/EdgeDeviceGetter.cs:31]: Do we want to have a work item to keep this under radar?

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs [hunk 2]
```diff
@@ -124,6 +126,15 @@ public void ConfigureServices(IServiceCollection services)
 
             if (NetworkServerConfiguration.ClientCertificateMode is not ClientCertificateMode.NoCertificate)
                 _ = services.AddSingleton<IClientCertificateValidatorService, ClientCertificateValidatorService>();
+
+            _ = NetworkServerConfiguration.RunningAsIoTEdgeModule
+                ? services.AddSingleton<ModuleConnectionHost>()
```
Comment (Mandur) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs:131]: nit: should we harmonize the name ModuleControlHost ? (Could be done later)

**Review by Mandur — APPROVED** [review]

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs (hunk 2 above)
Comment (danigian) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs:131]: Will keep as it is for now 👍

File: LoRaEngine/LoraKeysManagerFacade/EdgeDeviceGetter.cs (hunk 1 above)
Comment (danigian) [reply LoRaEngine/LoraKeysManagerFacade/EdgeDeviceGetter.cs:31]: We should not need a work item here. Ideally if a future version of the RegistryManager will support cancellation tokens, as soon as we upgrade the dependency, the build should fail as we are not propagating the cancellation token (as per https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2016)

**Pull request merged**